### PR TITLE
[SEASK-49] Develop starter kit extensions

### PR DESCRIPTION
**Pull request by semalaiappan**
StudentSchoolAssociation.ReportingSchool
StudentSchoolAssociation.ResidentLocalEducationAgency
StudentSchoolAssociation.ResidentSchool
StudentLanguageInstructionProgramAssociation.RedesignatedEnglishFluent
StudentSpecialEducationProgramAssociation.toTakeAlternateAssessment

All 5 Fields added in SK Extension

### CODE REVIEW

**Review by jamessmckay — CHANGES_REQUESTED** [review]
There are binary files that have been checked in, these need to be removed please

File: sample-extension/SkMetaEd/Association/StudentLanguageInstructionProgramAssociationExtension.metaed [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Association EdFi.StudentLanguageInstructionProgramAssociation additions
+bool RedesignatedEnglishFluent
+          documentation "students that have been redesignated as English Fluent."
+          is optional
```
Comment (tmeador-dl) [inline sample-extension/SkMetaEd/Association/StudentLanguageInstructionProgramAssociationExtension.metaed]:
```suggestion
Association EdFi.StudentLanguageInstructionProgramAssociation additions
    bool RedesignatedEnglishFluent
        documentation "Students that have been redesignated as English Fluent."
        is optional
```

File: sample-extension/SkMetaEd/Association/StudentSchoolAssociationExtension.metaed [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    is required
+    role name Resident
+domain entity EdFi.School
+    documentation "School within a district where student resides. If student resides out of state, use school of enrollment or leave blank."
+    is optional
+    role name Resident
+domain entity EdFi.School
+    documentation "School within a district where student's records should be reported for state reporting purposes."
+    is required
+    role name Reporting
```
Comment (tmeador-dl) [inline sample-extension/SkMetaEd/Association/StudentSchoolAssociationExtension.metaed]:
```suggestion
Association EdFi.StudentSchoolAssociation additions
    domain entity EdFi.LocalEducationAgency
        documentation "District where student resides. If student resides out of state, use district of enrollment."
        is required
        role name Resident
    domain entity EdFi.School
        documentation "School within a district where student resides. If student resides out of state, use school of enrollment or leave blank."
        is optional
        role name Resident
    domain entity EdFi.School
        documentation "School within a district where student's records should be reported for state reporting purposes."
        is required
        role name Reporting
```

File: sample-extension/SkMetaEd/Association/StudentSpecialEducationProgramAssociationExtension.metaed [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Association EdFi.StudentSpecialEducationProgramAssociation additions
+bool ToTakeAlternateAssessment
+    documentation "True = Indicates the student needs alternate assessment."
+    is optional
```
Comment (tmeador-dl) [inline sample-extension/SkMetaEd/Association/StudentSpecialEducationProgramAssociationExtension.metaed]:
```suggestion
Association EdFi.StudentSpecialEducationProgramAssociation additions
    bool ToTakeAlternateAssessment
        documentation "True = Indicates the student needs alternate assessment."
        is optional
```

**Comment by semalaiappan** [comment]
@jamessmckay Fixed your review comments .

**Review by jamessmckay — CHANGES_REQUESTED** [review]
These files that are in the obj folder should not be in the pr too
image
Also did you make the change to the .gitigore?

**Comment by semalaiappan** [comment]
added gitignore to fix the issue

File: sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    <description>EdFi.Ods.Extensions.Sk</description>
+    <releaseNotes></releaseNotes>
+    <license type="file">LICENSE.txt</license>
+  </metadata>
+  <files>
+    <file src="../../LICENSE.txt" />    
+    <file src="Artifacts/**/*.sql" target="."/>
+    <file src="Artifacts/**/*.json" target="."/>
+    <file src="assemblyMetadata.json" target="."/>
+    <file src="bin\$configuration$\netcoreapp3.1\EdFi.Ods.Extensions.Sk.dll" target="."/>
```
Comment (jamessmckay) [inline sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec]:
```suggestion
    <file src="bin/$configuration$/netcoreapp3.1/EdFi.Ods.Extensions.Sk.dll" target="."/>
```

File: .gitignore [hunk 5]
```diff
@@ -7,3 +8,45 @@
...
+
+# Build Folders (you can keep bin if you'd like, to store dlls and pdbs)
+[Bb]in
+csx/
+rcf/
+[Oo]bj
+logistics/bin
+.vs/
+.idea/
+
```
Comment (jamessmckay) [inline .gitignore:20]:
```suggestion
```

File: .gitignore [hunk 6]
```diff
@@ -7,3 +8,45 @@
...
+*.pubxml.user
+*.dbmdl
+.vs/
+# Resharper files
+_ReSharper.*
+*.resharper
+*.resharper.user
+
+
+
```
Comment (jamessmckay) [inline .gitignore]:
```suggestion
```

File: .gitignore [hunk 7]
```diff
@@ -7,3 +8,45 @@
...
+#Others
+*.orig
+*.original
+
+
+Starter-Kit-SEA-Data-Collection/sample-extension/EdFi.Ods.Extensions.Sk/bin/**/*.*
+Starter-Kit-SEA-Data-Collection/sample-extension/EdFi.Ods.Extensions.Sk/obj/**/*.*
```
Comment (jamessmckay) [inline .gitignore]:
```suggestion
```

Note the bin/ and obj/ does this

**Review by jamessmckay — CHANGES_REQUESTED** [review]
some minor changes

**Review by jamessmckay — APPROVED** [review]

**Review by jamessmckay — APPROVED** [review]

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**